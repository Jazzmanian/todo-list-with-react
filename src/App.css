* {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}
.complete {
  text-decoration: line-through;
}

.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #141a2f;
}
.title {
  color: white;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: auto;
  margin-bottom: 0rem;
  margin-top: 0rem;
}

input {
  border: 0;
  outline: 0;
  padding: 5px;
  border: solid 4px rgb(248, 91, 0);
  background-color: #141a2f;
  color: white;
  border-top-left-radius: 25px;
  border-bottom-left-radius: 25px;
}
.form button {
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  background-color: rgb(248, 91, 0);
  border: 0;
  outline: 0;
  color: white;
  padding: 9px;
}
.elements {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 2rem;
}
.elements div {
  margin-left: 2rem;
  margin-right: 2rem;
  margin-top: 1rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.7rem;
  border-radius: 0.3rem;
  background: hsla(186, 100%, 69%, 1);
  color: rgb(255, 255, 255);
  background: hsla(217, 100%, 50%, 1);

  background: linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -moz-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -webkit-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  filter: progid: DXImageTransform.Microsoft.gradient( startColorstr="#0061FF", endColorstr="#60EFFF", GradientType=1 );
}
