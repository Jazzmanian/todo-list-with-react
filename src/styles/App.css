* {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}
.complete {
  text-decoration: line-through;
}

.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #141a2f;
}
.title {
  color: white;
  margin-bottom: 1.5rem;
  margin-top: 1.5rem;
}

.form {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-bottom: auto;
  margin-bottom: 0rem;
  margin-top: 0rem;
}

input {
  border: 0;
  outline: 0;
  padding: 5px;
  border: solid 4px rgb(248, 91, 0);
  background-color: #141a2f;
  color: white;
  border-top-left-radius: 25px;
  border-bottom-left-radius: 25px;
  margin-left: 60px;
}
.form button {
  border-top-right-radius: 25px;
  border-bottom-right-radius: 25px;
  background-color: rgb(248, 91, 0);
  border: 0;
  outline: 0;
  color: white;
  padding: 9px;
}
.elements {
  display: flex;
  align-items: center;
  flex-direction: column;
  margin-top: 2rem;
}
.todo-items {
  margin-left: 2rem;
  margin-right: 2rem;
  margin-top: 1rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.7rem;
  border-radius: 0.3rem;
  background: hsla(186, 100%, 69%, 1);
  color: rgb(255, 255, 255);
  background: hsla(217, 100%, 50%, 1);

  background: linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -moz-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  background: -webkit-linear-gradient(
    0deg,
    hsla(217, 100%, 50%, 1) 0%,
    hsla(186, 100%, 69%, 1) 100%
  );

  filter: progid: DXImageTransform.Microsoft.gradient( startColorstr="#0061FF", endColorstr="#60EFFF", GradientType=1 );
}
.todo-items h4 {
  padding-right: 10rem;
}
.todo-items button {
  font-size: 120%;
  background-color: inherit;
  border: 0;
}
