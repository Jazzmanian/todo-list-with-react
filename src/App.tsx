import React from 'react';
import TodoForm from './components/TodoForm';
import TodoList from './components/TodoList';
import { useToggle } from './hooks/useToggle';
import { ThemeProvider } from 'styled-components';
import { GlobalStyles } from './styles/Global';
import { TodoContainer } from './styles/Container.styles';
import { theme } from './styles/theme';

const App: React.FC = () => {
  const { data, addTodo, handleDelete, toggleComplete } = useToggle();

  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      <TodoContainer>
        <TodoForm addTodo={addTodo} />
        <TodoList
          taskList={data}
          handleDelete={handleDelete}
          toggleComplete={toggleComplete}
        />
      </TodoContainer>
    </ThemeProvider>
  );
};

export default App;
