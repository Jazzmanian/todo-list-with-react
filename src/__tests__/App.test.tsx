import { render, screen } from '@testing-library/react';
import App from '../App';
import React from 'react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event';

describe('App test', () => {
  it('should render app', () => {
    render(<App />);
    expect(screen.getByText(/Todo list/i)).toBeInTheDocument();
  });
  it('should render the snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should update state when a new item is received from input box', () => {
    render(<App />);
    const button = screen.getByText(/add items/i);
    const textInput = screen.getByRole('textbox');
    userEvent.type(textInput, 'task 01');
    userEvent.click(button);
    expect(screen.getByText('task 01')).toBeInTheDocument();
  });
  it('should update id when a new item is received from input box', () => {
    render(<App />);
    const button = screen.getByText(/add items/i);
    const textInput = screen.getByRole('textbox');
    userEvent.type(textInput, 'task 01');
    userEvent.click(button);
    userEvent.type(textInput, 'task 02');
    userEvent.click(button);
    expect(screen.getByTestId(1).textContent).toBe('task 01');
    expect(screen.getByTestId(2).textContent).toBe('task 02');
  });
});
