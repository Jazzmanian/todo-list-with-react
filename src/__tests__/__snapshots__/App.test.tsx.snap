// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should render the snapshot 1`] = `
<div
  className="sc-eDvSVe sc-pyfCe jSrjjn fvMWM"
>
  <div>
    <h1
      className="sc-gswNZR jSLNOy"
    >
      Todo List
    </h1>
    <input
      aria-label="todo-input"
      autoFocus={true}
      className="sc-dkrFOg bQFNLL"
      onChange={[Function]}
      placeholder="Enter your todo item"
      type="text"
      value=""
    />
    <button
      className="sc-bcXHqe sc-hLBbgP htysNs hSIAgy"
      onClick={[Function]}
    >
      Add Items
    </button>
  </div>
  <div
    className="sc-eDvSVe sc-jSUZER jSrjjn gAoPZC"
  />
</div>
`;
